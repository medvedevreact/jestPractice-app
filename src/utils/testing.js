import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

export const renderWithRouter = (component, options = {}) => {
  const Wrapper = ({ children }) => (
    <MemoryRouter {...options}>{children}</MemoryRouter>
  );
  return render(component, { wrapper: Wrapper });
};
