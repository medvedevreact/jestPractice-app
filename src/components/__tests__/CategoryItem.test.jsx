import { screen } from '@testing-library/react';
import { CategoryItem } from '../CategoryItem';
import { renderWithRouter } from '../../utils/testing';

describe('CategoryItem component', () => {
  const categoryMock = {
    strCategory: 'Dessert',
    strCategoryThumb: '/dessert.png',
    strCategoryDescription:
      'A variety of delicious desserts from around the world.',
  };

  it('renders correctly', () => {
    renderWithRouter(<CategoryItem {...categoryMock} />);
    expect(screen.getByRole('article')).toMatchSnapshot();
  });
});
