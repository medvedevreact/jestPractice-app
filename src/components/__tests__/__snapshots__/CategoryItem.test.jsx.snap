// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem component renders correctly 1`] = `
<div
  class="card"
  role="article"
>
  <div
    class="card-image"
  >
    <img
      alt="Dessert"
      src="/dessert.png"
    />
  </div>
  <div
    class="card-content"
  >
    <span
      class="card-title"
    >
      Dessert
    </span>
    <p>
      A variety of delicious desserts from around the world.
      ...
    </p>
  </div>
  <div
    class="card-action"
  >
    <a
      class="btn"
      href="/category/Dessert"
    >
      Watch category
    </a>
  </div>
</div>
`;
